app/(interview)/onboard/[protocolId]/route.ts

```diff
@@ -13,6 +15,15 @@ const handler = async (
     return NextResponse.redirect(new URL('/onboard/error', req.nextUrl));
   }
 
+  const appSettings = await api.appSettings.get.query();
+
+  // if limitInterviews is enabled
+  // Check cookies for interview already completed for this user for this protocol
+  // and redirect to finished page
+  if (appSettings?.limitInterviews && cookies().get(protocolId)) {
+    redirect('/interview/finished');
+  }
```

Comment:
In future, think about making this a single function in a separate file, that the route handler and the page can both import. That way we can update the behaviour later in a simpler way.